plugins/gatsby-source-install-config/gatsby-node.js

```diff
@@ -0,0 +1,159 @@
+exports.createSchemaCustomization = ({ actions }) => {
+  const { createTypes } = actions;
+
+  createTypes(`
+    type InstallConfig @dontInfer {
+      id: ID!
+      agentName: String!
+      agentType: String!
+      title: String!
+      intro: IntroFile
+      appInfo: [AppInfoOption!]!
+      steps: [InstallStep]
+      mdxFiles: [Mdx]
+      whatsNextMdx: String!
+    }
+    type IntroFile @dontInfer {
+      filePath: String
+      mdx: Mdx
+    }
+    type AppInfoOption @dontInfer {
+      optionType: String!
+      label: String!
+      options: [AppInfoOptionValue!]!
+    }
+    type AppInfoOptionValue @dontInfer {
+      value: String!
+      displayName: String!
+      recommendedGuided: Boolean
+    }
+    type InstallStep @dontInfer {
+      filePath: String
+      mdx: Mdx
+      overrides: [StepOverride]
+    }
+    type StepOverride @dontInfer {
+      optionType: String!
+      overrideConfig: [StepOverrideConfig]
+    }
+    type StepOverrideConfig @dontInfer {
+      value: String!
+      filePath: String
+      mdx: Mdx
+      skip: Boolean
+      overrides: [StepOverride]
+    }
+    type Mdx implements Node {
+      frontmatter: Frontmatter
+    }
+    type Frontmatter {
+      componentType: String! 
+      headingText: String
+      inputOptions: [InputOption]
+      tipMdx: String
+      agentConfigFilePath: String
+      optionType: String
+    }
+    type InputOption @dontInfer {
+      labelMdx: String!
+      value: String! 
+      infoMdx: String! 
+    }
+  `);
+};
+
+exports.createResolvers = ({ createResolvers, createNodeId }) => {
+  createResolvers({
+    Query: {
+      installConfig: {
+        type: 'InstallConfig',
+        args: {
+          agentName: 'String!',
+        },
+        resolve: async (_source, args, context) => {
+          const { agentName } = args;
+          const { nodeModel } = context;
+          const { entries } = await nodeModel.findAll({ type: 'ConfigYaml' });
```

Comment:
do we want to use `nodeModel.findOne()` on the type `configYaml` insted so we can filter by agentName ahead of time instead of getting/loading all YAML data and running `.find()`?  not sure how much performance gains there are really are. ¯\_(ツ)_/¯

with [`findOne()` ](https://www.gatsbyjs.com/docs/reference/graphql-data-layer/node-model/#findOne) you could filter for the `agentName`:

```
const { entries } = await nodeModel.findOne({
  type: "ConfigYaml",
  query: {
    filter: {
      locale: { eq: agentName },
    },
  },
});
```